llvm/test/CodeGen/RISCV/rvv/fixed-vectors-interleaved-access-zve32x.ll

```diff
@@ -7,14 +7,14 @@
 define <4 x i1> @load_large_vector(ptr %p) {
 ; ZVE32X-LABEL: load_large_vector:
 ; ZVE32X:       # %bb.0:
-; ZVE32X-NEXT:    ld a1, 80(a0)
-; ZVE32X-NEXT:    ld a2, 72(a0)
-; ZVE32X-NEXT:    ld a3, 56(a0)
-; ZVE32X-NEXT:    ld a4, 32(a0)
-; ZVE32X-NEXT:    ld a5, 24(a0)
-; ZVE32X-NEXT:    ld a6, 48(a0)
-; ZVE32X-NEXT:    ld a7, 8(a0)
-; ZVE32X-NEXT:    ld a0, 0(a0)
+; ZVE32X-NEXT:    ld a1, 8(a0)
```

Comment:
These offsets become more irregular (though they were not so regular before). Maybe this can be a future optimization?

Reply:
I'm not sure why this combine is specifically triggering these loads to be come unordered, but https://github.com/llvm/llvm-project/pull/73789 may fix this

Reply:
I just tried it, the result is:
```
; ZVE32X-NEXT:    ld a1, 48(a0)
; ZVE32X-NEXT:    ld a2, 56(a0)
; ZVE32X-NEXT:    ld a3, 72(a0)
; ZVE32X-NEXT:    ld a4, 80(a0)
; ZVE32X-NEXT:    ld a5, 0(a0)
; ZVE32X-NEXT:    ld a6, 8(a0)
; ZVE32X-NEXT:    ld a7, 24(a0)
; ZVE32X-NEXT:    ld a0, 32(a0)
```
It seems that elements in subvectors are ordered, but it's not the case for subvectors.

Reply:
After limiting the splitting down to MF2 the original order of these loads seems to be restored

Reply:
Thanks! I don't know which one is more worthwhile. Maybe we can loose the limitation in the future. :-)